applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing_matrix_free.h

```diff
@@ -142,42 +142,53 @@ class MapperVertexMorphingMatrixFree : public Mapper
         mValuesDestination[2].clear();
 
         // Perform mapping
-        for(auto& node_i : mrDestinationModelPart.Nodes())
+        const auto destination_nodes_begin = mrDestinationModelPart.NodesBegin();
+
+        #pragma omp parallel for firstprivate(destination_nodes_begin)
```

Comment:
I am not sure about the `firstprivate(destination_nodes_begin)`. As far as i understand the definition of it, it is not necessary here.